### ISSUE TITLE
GULSceneDelegateSwizzler.h not found

### ISSUE THREAD

**Issue by raulrom13** [firebase/firebase-ios-sdk#4684]
### Describe your environment

  * Xcode version: 11.3.1
  * Firebase SDK version: 6.14.0
  * Firebase Component: Auth
  * Component version: 6.4.2

### Describe the problem

#import <GoogleUtilities/GULSceneDelegateSwizzler.h>

Pods/FirebaseAuth/Firebase/Auth/Source/Auth/FIRAuth.m:32:9: 'GoogleUtilities/GULSceneDelegateSwizzler.h' file not found

#### Steps to reproduce:

Pod update to Firebase/Auth 6.4.2

**Comment by rizafran** [comment]
Hi @raulrom13, thanks for reporting this issue. Looks like the `GULSceneDelegateSwizzler.h` was just added to Firebase/Auth 6.4.2 (see [here](https://github.com/firebase/firebase-ios-sdk/pull/4524)). Could you try running the `pod update` command and see if the issue persists?

**Comment by raulrom13** [comment]
Yes, the issue persist with a pod update command.

I have seen that the issue is inside GoogleUtilities podspec. I have watchOS with platform 4.0, and in GoogleUtilities there are a minimum platform 6.0.

In my podspec, I'm using Firebase just in iOS dependencies, however Cocoapods doesn't update Firebase to 6.15.0 or GoogleUtilities to 6.5.0. If I change the watchOS version to 6.0, then Cocoapods updates all the firebase dependencies.

`"name": "GoogleUtilities",
  "version": "6.5.0",
  "summary": "Google Utilities for iOS (plus community support for macOS and tvOS)",
  "description": "Internal Google Utilities including Network, Reachability Environment, Logger and Swizzling for\nother Google CocoaPods. They're not intended for direct public usage.",
  "homepage": "https://github.com/firebase/firebase-ios-sdk/tree/master/GoogleUtilities",
  "license": {
    "type": "Apache",
    "file": "LICENSE"
  },
  "authors": "Google, Inc.",
  "source": {
    "git": "https://github.com/firebase/firebase-ios-sdk.git",
    "tag": "Utilities-6.5.0"
  },
  "platforms": {
    "ios": "8.0",
    "osx": "10.11",
    "tvos": "10.0",
    "watchos": "6.0"
  }`

**Comment by rizafran** [comment]
@raulrom13, yes, I've seen [here](https://github.com/firebase/firebase-ios-sdk/blob/master/GoogleUtilities.podspec) that the deployment target for watchOS is 6.0.

Hi @ryanwilson, any thoughts on this?

**Comment by Aresgod112** [comment]
I am having the same issue and I am learning flutter and dart. 

`Running pod install...
Running Xcode build...
Xcode build done.                                           126.5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/mac/AndroidStudioProjects/flutter_shop/ios/Pods/FirebaseAuth/Firebase/Auth/Source/Auth/FIRAuth.m:32:9: fatal error: 'GoogleUtilities/GULSceneDelegateSwizzler.h' file not found
    #import <GoogleUtilities/GULSceneDelegateSwizzler.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/mac/AndroidStudioProjects/flutter_shop/ios/Pods/FirebaseAuth/Firebase/Auth/Source/Auth/FIRAuth.m:32:9: note: did not find header 'GULSceneDelegateSwizzler.h' in framework 'GoogleUtilities' (loaded from '/Users/mac/AndroidStudioProjects/flutter_shop/build/ios/Debug-iphonesimulator/GoogleUtilities')
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11 Pro Max.`

**Comment by ryanwilson** [comment]
Apologies everyone for the issues - @Aresgod112 can you confirm that you also have a watchOS application in your Podfile?

In the meantime while we sort this out, you can lock to an older version of Auth in your Podfile with:
`pod 'FirebaseAuth', '6.4.1'`

An immediate thing that sticks out is the [spec on CocoaPods shows a dependency on GoogleUtilities 6.2](https://github.com/CocoaPods/Specs/blob/bb847897c15f94846d87a0eff2fad43128cc871b/Specs/6/3/6/FirebaseAuth/6.4.2/FirebaseAuth.podspec.json#L47) but [the local spec declares a dependency on GoogleUtilities 6.5](https://github.com/firebase/firebase-ios-sdk/blob/2328d1f365982bccd37c92656f41f177814ea1c2/FirebaseAuth.podspec#L45) which is the proper version.

**Comment by raulrom13** [comment]
Yes, I have a watchOS target in my Podfile.

I tried this:

`ioscore.ios.dependency 'Firebase/Auth', '6.4.1'`

And I got that error:

`[!] CocoaPods could not find compatible versions for pod "Firebase/Auth":
  In Podfile:
    SkitudeCore (from `../`) was resolved to 1006, which depends on
      SkitudeCore/iOSCore (= 1006) was resolved to 1006, which depends on
        Firebase/Auth (~> 6.4.1)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Auth (~> 6.4.1)`.

You have either:
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.`

**Comment by raulrom13** [comment]
I have changed:
`'Firebase/Auth', '6.4.1'`
for:
`'FirebaseAuth', '6.4.1'`

And now all is working again.

**Comment by ryanwilson** [comment]
Beat me to it - was just about to comment to remove the `/`. Glad it's working, I think I found out the root cause and I'll post about it shortly after I type it up. Will need to find the right approach to resolve things as well.

**Comment by raulrom13** [comment]
Thank you @ryanwilson, I will try it again when you fix it.

**Comment by ryanwilson** [comment]
The root cause of the issue:

Updates to the podspecs were cherry-picked to the release in #4640 and the associated tags were updated, but our tooling failed to push all podspecs to our staging area since there were inter-pod dependencies and it tried to push out of order. We manually pushed the podspecs to the staging area but missed `FirebaseAuth.podspec` and `FirebaseCrashlytics.podspec`. 

The changes were as follows:

`FirebaseAuth`:
  - `FirebaseCore` dependency should be `6.6` but it stayed on `6.2`
  - `GoogleUtilities` dependency should be `6.5` but it stayed on `6.2`
`FirebaseCrashlytics:`
  - `FirebaseCore` dependency should be `6.6` but it stayed on `6.3.2`
  - `FirebaseInstanceID` dependency should be `4.3` but it stayed on `4.1`

We're working on a resolution here but will have to wait a bit until the rest of folks get online (it's still early on the East coast). 

One other thing we didn't foresee: adding watchOS support to `FirebaseMessaging` and all dependencies (including `FirebaseCore` and `GoogleUtilities`) ended up preventing you from being able to update to the latest pods even though you aren't using it in your older watchOS app.  If possible, can you please share your Podfile? You can remove non-Firebase pods if you want but please keep the structure and any watchOS related items. I want to see if we can tell CocoaPods that the watchOS version should be limited to your watchOS target and not the iOS target as well, allowing you to upgrade to the latest version of `FirebaseCore` and deps.

**Comment by raulrom13** [comment]
**Podfile**:

`
def shared_pods
  #inhibit_all_warnings!
  pod 'SKCCore', :path => '../'
end

# iOS target
target 'XX' do
  platform :ios, '11.0'
  shared_pods
end

# SkitudeWatch target
target 'XX' do
  platform :watchos, '4.0'
  shared_pods
end
`

**Core Podspec**:
`
...
spec.subspec 'iOSCore' do |ioscore|
...
  ioscore.ios.dependency 'FirebaseAnalytics'
  ioscore.ios.dependency 'FirebaseAuth', '6.4.1'
  ioscore.ios.dependency 'FirebaseMessaging'
  ioscore.ios.dependency 'FirebasePerformance'
...
end

spec.subspec 'WatchCore' do |watchcore|
... // No Firebase dependency
end
`

**Comment by ryanwilson** [comment]
@raulrom13 can you please test something for me that will help confirm the issue?

Can you replace:
```
  ioscore.ios.dependency 'FirebaseAnalytics'
  ioscore.ios.dependency 'FirebaseAuth', '6.4.1'
  ioscore.ios.dependency 'FirebaseMessaging'
  ioscore.ios.dependency 'FirebasePerformance'
```

with 
```
  ioscore.ios.dependency 'Firebase/Analytics', '6.15.0'
  ioscore.ios.dependency 'Firebase/Auth'
  ioscore.ios.dependency 'Firebase/Messaging'
  ioscore.ios.dependency 'Firebase/Performance'
```

I'm hoping to see what the exact error is that's preventing Firebase from being updated to `6.15.0` (which includes `FirebaseCore` `6.6` and others).

Thanks!

**Comment by Aresgod112** [comment]
@ryanwilson To be honest all I was doing was trying to learn to create a simple ios login (not for watch ) app with firebase.
 in my pubspec.yaml file the dependencies for firebase are   
  cloud_firestore: ^0.13.0+1
  firebase_auth: ^0.15.3+1.

I don't know where to look for those files you guys are talking about. Sorry , as I said I am very new to this but I am trying my best to learn. So please if you would be kind to give me a bit of more guidance, I would appreciate it .

**Comment by raulrom13** [comment]
@ryanwilson this is the error with your configuration.

`Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "FirebaseAnalytics":
  In Podfile:
    SkitudeCore (from `../`) was resolved to 1006, which depends on
      SkitudeCore/iOSCore (= 1006) was resolved to 1006, which depends on
        FirebaseAnalytics (= 6.15.0)

None of your spec sources contain a spec satisfying the dependency: `FirebaseAnalytics (= 6.15.0)`.

You have either:
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
`

**Comment by ryanwilson** [comment]
Thanks for the reply - did you add the slash back in? i.e `Firebase/Analytics` instead of `FirebaseAnalytics`.

**Comment by ryanwilson** [comment]
> @ryanwilson To be honest all I was doing was trying to learn to create a simple ios login (not for watch ) app with firebase.
> in my pubspec.yaml file the dependencies for firebase are
> cloud_firestore: ^0.13.0+1
> firebase_auth: ^0.15.3+1.
> 
> I'm working in Android studio , Dart Flutter .
> I created just a login page for ios and also connected it to firebase. But when I want to launch the app on the simulator it gives me the above problem .
> I don't know where to look for those files you guys are talking about. Sorry , as I said I am very new to this but I am trying my best to learn. So please if you would be kind to give me a bit of more guidance, I would appreciate it .

No worries - thanks for any info you can provide! I'm not familiar with building Flutter apps myself - is there a tutorial you're following that I could try to replicate or some series of steps I could follow and try it myself?

**Comment by Aresgod112** [comment]
> > @ryanwilson To be honest all I was doing was trying to learn to create a simple ios login (not for watch ) app with firebase.
> > in my pubspec.yaml file the dependencies for firebase are
> > cloud_firestore: ^0.13.0+1
> > firebase_auth: ^0.15.3+1.
> > I'm working in Android studio , Dart Flutter .
> > I created just a login page for ios and also connected it to firebase. But when I want to launch the app on the simulator it gives me the above problem .
> > I don't know where to look for those files you guys are talking about. Sorry , as I said I am very new to this but I am trying my best to learn. So please if you would be kind to give me a bit of more guidance, I would appreciate it .
> 
> No worries - thanks for any info you can provide! I'm not familiar with building Flutter apps myself - is there a tutorial you're following that I could try to replicate or some series of steps I could follow and try it myself?

Thank you for your answer . Sure  here is the link.

**Comment by raulrom13** [comment]
Sorry @ryanwilson, the new error message:

``
[!] CocoaPods could not find compatible versions for pod "FirebaseCore":
  In Podfile:
    SkitudeCore (from `../`) was resolved to 1006, which depends on
      SkitudeCore/iOSCore (= 1006) was resolved to 1006, which depends on
        Firebase/Analytics (= 6.15.0) was resolved to 6.15.0, which depends on
          Firebase/Core (= 6.15.0) was resolved to 6.15.0, which depends on
            Firebase/CoreOnly (= 6.15.0) was resolved to 6.15.0, which depends on
              FirebaseCore (= 6.6.0)

    SkitudeCore (from `../`) was resolved to 1006, which depends on
      SkitudeCore/iOSCore (= 1006) was resolved to 1006, which depends on
        Firebase/Analytics (= 6.15.0) was resolved to 6.15.0, which depends on
          Firebase/Core (= 6.15.0) was resolved to 6.15.0, which depends on
            FirebaseAnalytics (= 6.2.1) was resolved to 6.2.1, which depends on
              FirebaseCore (~> 6.6)

Specs satisfying the `FirebaseCore (= 6.6.0), FirebaseCore (~> 6.6)` dependency were found, but they required a higher minimum deployment target.
``

**Comment by ryanwilson** [comment]
@raulrom13 I think this could potentially be a CocoaPods bug due to the complex setup in your Podfile that's preventing you from updating to the latest version.

I was able to get a simple project that has watchOS 4.0 project working when Firebase was included, bringing in the latest version of the SDK:

```
platform :ios, '9.0'

target 'Blank' do
  use_frameworks!
  pod 'Firebase/Analytics'
  pod 'Firebase/Messaging'
end

target 'WatchWatchWatch' do
  use_frameworks!
  platform :watchos, '4.0'
end
```

I'm curious, what are the default subspecs you have for your `SKCCore` pod? I assume the `iOSCore` subspec is being pulled into the watchOS target as well, and even though those pods aren't included for that platform maybe that happens *after* the version resolution step in CocoaPods.

A quick way to test this would be to only use the necessary subspecs for your watchOS target:

```
target 'XX' do
  platform :watchos, '4.0'
  pod 'SKCCore/WatchCore', :path => '../'
  # Also any other subspecs that are included by default in `SKCCore`...
end
```

@Aresgod112 I've asked a colleague who is more familiar with Flutter to look into this, but if they can't I'll get back to it shortly.

**Comment by raulrom13** [comment]
wow @ryanwilson  I tried your recommendation and now all is working like a charm.

```
# Watch target
target 'WatchExtension' do
  platform :watchos, '4.0'
  pod 'xxxCore/watchOSCore', :path => '../'
end
```

Thank you! 💯

**Comment by ryanwilson** [comment]
@raulrom13 happy to hear! You're very welcome. Sounds like you hit an edge case with CocoaPods, I'll see if I can get a small repro that I can file a bug with them. Cheers!

I won't close this bug yet until we figure out the problem @Aresgod112 is seeing.

**Comment by Aresgod112** [comment]
@ryanwilson Much appreciated . I'm stuck for 2 days on this and I've searched all over . I've seen this problem in other places as well but no answer .

**Comment by ryanwilson** [comment]
@Aresgod112 in the meantime as a workaround (so we can unblock you), can you please try an older version of Firebase? For `firebase_auth` the are here: https://pub.dev/packages/firebase_auth/versions so something like `0.15.3`, `0.15.2`, and moving backwards.  You'll likely need to match `cloud_firestore` as well, `0.12.11` or `0.12.10+2`

**Comment by Aresgod112** [comment]
I'm at   firebase_auth: ^0.15.2
  cloud_firestore: ^0.12.10+2      still getting the same error , sorry it takes some time cause every time I have to clean the build.  Tried all the combinations until now and including this one still same error.

**Comment by ryanwilson** [comment]
Hmm we aren't able to reproduce it on our end. Can you please try: `pod repo update` to make sure you have all the latest SDKs available, and copy the contents of your `Podfile.lock`? It should be in the `ios` directory. That'll tell us which versions are being pulled in (moving back to the original versions specified below).
```
cloud_firestore: ^0.13.0+1
firebase_auth: ^0.15.3+1.
```

**Comment by Aresgod112** [comment]
But when I do

`pod update` it gives me this 

[!] No `Podfile' found in the project directory.
MacdeMacBook-Air:flutter_shop1 mac$

**Comment by Aresgod112** [comment]
> Hi.
> I face same problem.
> And `pod update` on flutter/ios solved in my case.
> This was helpful.
> https://stackoverflow.com/questions/59754797/googleutilities-gulscenedelegateswizzler-h-file-not-found
> 
> * `flutter clean` - can't solve
> * pubspec.yaml change older version firebase_auth - can't solve
> * `pod repo update`* + `pod update` - solve.
> 
> I think flutter setup flow not support cocoapod's trash clean.

This worked for me as well. Thank you. 
In my case, I had to execute this command from `AndoidStudioProjects/flutter_app_name/ios/` for anybody else who tried to do that from )


Now I am having another issue after fixing this. 

`haseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/mac/Library/Developer/Xcode/DerivedData/Runner-fgxkoxxcwtxceifctixthvenvsrs/Build/Intermediates.noindex/Runner.build/Debug-iphonesimulator/Runner.build/Script-763CFB666E743EDD9491ECFC.sh (in target 'Runner' from project 'Runner')
    cd /Users/mac/AndroidStudioProjects/flutter_shop1/ios
    /bin/sh -c /Users/mac/Library/Developer/Xcode/DerivedData/Runner-fgxkoxxcwtxceifctixthvenvsrs/Build/Intermediates.noindex/Runner.build/Debug-iphonesimulator/Runner.build/Script-763CFB666E743EDD9491ECFC.sh

/Users/mac/Library/Developer/Xcode/DerivedData/Runner-fgxkoxxcwtxceifctixthvenvsrs/Build/Intermediates.noindex/Runner.build/Debug-iphonesimulator/Runner.build/Script-763CFB666E743EDD9491ECFC.sh: line 2: /Users/mac/AndroidStudioProjects/flutter_shop1/ios/Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh: Permission denied
Command PhaseScriptExecution failed with a nonzero exit code
` 

but I guess for this I should open a new ticket. 

Thank you all @ryanwilson, @takuro-kira, @raulrom13 for the help.

**Comment by ryanwilson** [comment]
Happy to hear the original issue is resolved, and thanks @takuro-kira for the specific instructions!

Yes, @Aresgod112 that doesn't look like a Firebase specific issue but try cleaning your derived data in Xcode and see if that helps. If you see any other Firebase specific issues please open another issue and we'll do our best to resolve it!
